arbos/burn/burn.go

```diff
@@ -13,6 +13,7 @@ import (
 type Burner interface {
 	Burn(amount uint64) error
 	Burned() uint64
+	GasLeft() *uint64
```

Comment:
I'd add a comment here saying that it'll panic for the system burner. As-is it's a potential footgun IMO.